### PR TITLE
Removed unused dependency

### LINKED ISSUE #227: [FEATURE] Publish box at maven or merge it in apalache
The `box` dependency is the only external non-maven dependency that is left in the project. #219 turns z3 into a maven dependency, tla2tools are imported by maven as well. If we find a way to mavenize `box`, we will eliminate the need for `3rdparty/install-local.sh`altogether.

@Kukovec, since Box is your library, I see two solutions:

 1. You maintain the library and publish it at maven,
 1. You merge the library in apalache, so it is not an external dependency anymore. You can make it a separate module with its own `pom.xml`, so the users will be able to pull it as a separate jar.

As far as I can tell, you are not actively developing this library. So I would prefer solution 2, as it would make its maintenance less pain for everybody.

This is a critical issue related to the infrastructure, as it is preventing us from publishing Apalache modules at Maven Central.

### PR DESCRIPTION
**Pull request by Kukovec**
Closes #227.

### CODE REVIEW

**Review by konnov — APPROVED** [review]
Really cool that all of our dependencies are in maven now!

**Review by shonfeder — COMMENTED** [review]
I think the changes in #236 should be integrated too. I'm building the dockerfile now, to see if it's all still good :)

File: .travis.yml [hunk 1]
```diff
@@ -21,14 +21,8 @@ addons:
 
 cache:
   directories:
-    - 3rdparty/z3
```
[the remark this answers is not in the document]
Comment (Kukovec) [reply .travis.yml:24]: Hmm, good point, I'll try to remove them.

**Comment by shonfeder** [comment]
The build of the Dockerfile failed.

I'm happy to work on the followup in #236 as an independent followup (and it should also include putting the Dockerfile under CI).

**Comment by Kukovec** [comment]
I've cleaned the travis file, lets resolve the docker issue in #236.

**Comment by shonfeder** [comment]
LGTM!

**Review by shonfeder — APPROVED** [review]

**Pull request merged**